sources/packages/backend/workflow/test/2024-2025/assessment-gateway/assessment-gateway.originalAssessment.e2e-spec.ts

```diff
@@ -87,6 +87,106 @@ describe(`E2E Test Workflow assessment gateway on original assessment for ${PROG
     );
   });
 
+  it("Should update NOA status to not required when the application is completed.", async () => {
+    // Arrange
+
+    // Assessment consolidated mocked data.
+    const assessmentConsolidatedData: AssessmentConsolidatedData = {
+      assessmentTriggerType: AssessmentTriggerType.OriginalAssessment,
+      ...createFakeConsolidatedFulltimeData(PROGRAM_YEAR),
+      ...createFakeSingleIndependentStudentData(),
+      // Application with PIR not required.
+      studentDataSelectedOffering: 1,
+    };
+
+    assessmentConsolidatedData.applicationStatus = ApplicationStatus.Completed;
+
+    const workersMockedData = createWorkersMockedData([
+      createLoadAssessmentDataTaskMock({ assessmentConsolidatedData }),
+      createVerifyApplicationExceptionsTaskMock(),
+      createIncomeRequestTaskMock({
+        incomeVerificationId: incomeVerificationId++,
+        subprocesses: WorkflowSubprocesses.StudentIncomeVerification,
+      }),
+      createVerifyAssessmentCalculationOrderTaskMock(),
+    ]);
+
+    const currentAssessmentId = assessmentId++;
+
+    // Act/Assert
```

Comment:
Differently from the API ones the "Act" and "Assert" are separated here.
The "Assert" section starts at line 126.

Reply:
Updated

Reply:
Please review the similar cases in the new methods added.